### ISSUE TITLE
spec violation: springfox generates json without parameter type for @RequestParam Map<String, String>

### ISSUE THREAD

**Issue by nikit-cpp** [springfox/springfox#1554]
- What version of the library are you using? Is it the latest version?

2.6.0

 - What kind of issue is this?

Bug report. 

I start from example from [here](https://github.com/Swagger2Markup/spring-swagger2markup-demo).
If my businness method is something like
```java
fail(@RequestParam Map<String, String> requestParameters)
```
then Swagger2Markup (component of both swagger2markup gradle or maven plugin) fails on Assertion, see stacktrace [here](http://pastebin.com/RA5iSrE2).

[Here](https://github.com/Swagger2Markup/swagger2markup/blob/master/src/main/java/io/github/swagger2markup/internal/type/BasicType.java#L41) is validation in Swagger2Markup which fails.

I investigates it and in [OpenApi specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameter-object) said that:

> If **in** is any value other than "body": ... **field 'type' requiered**

I get swagger's json (from test below) and prettify it via jsonlint, and in it json has no 'type' parameter.
![jsonlint](http://i.imgur.com/sgje3Lm.png)
... but 'items' are present, may be it's attempt to generate array?

I wrote some test for it based on boot-static-docs, which check existence of 'type' parameter, if need I can write full example project with swagger2markup gradle plugin.

[Here](https://github.com/nikit-cpp/springfox-demos/blob/master/boot-static-docs/src/test/groovy/springfoxdemo/staticdocs/StaticDocsTest.groovy#L73) is test

[Here](https://github.com/nikit-cpp/springfox-demos/blob/master/boot-static-docs/src/main/java/springfoxdemo/staticdocs/rest/TestController.java) is controller

**Comment by dilipkrish** [comment]
See [here](https://github.com/swagger-api/swagger-ui/issues/2241#issuecomment-245123577) for more details.

TL;DR is that, yes this is a known violation, and the choice is to NOT use *non-compliant* operation implementations or live with this issue. At the end of the day, the spec is of no use if it cannot document what you're doing in real life.

None of the prescribed type values are suitable for _this_ scenario, and from a type specification perspective this is accurate. This will most likely also be changing in 3.0 of the spec, so in the mean time, I would ask @RobWin to consider handling this scenario in swagger2markup.

**Comment by RobWin** [comment]
As far as I remember Swagger2Markup supports Maps since this PR: https://github.com/Swagger2Markup/swagger2markup/pull/121/

**Comment by dilipkrish** [comment]
@RobWin thanks for checking on this!

**Comment by nikit-cpp** [comment]
@RobWin 
@dilipkrish 
Great =),
but there is stacktrace http://pastebin.com/RA5iSrE2
may be I should write proof of concept(project with gradle, swagger2markup, ...) which demonstrate bug ?

**Comment by RobWin** [comment]
Could you please provide a Swagger JSON/YAML file so that I can reproduce it?
No need for a full project.

**Comment by nikit-cpp** [comment]
@RobWin 
https://gist.github.com/nikit-cpp/f5cbd3ab748ec995cc93f6c5524322c2#file-swagger-json-L923

**Comment by RobWin** [comment]
We could only try the type: array.
The type is indeed mandatory.

**Comment by RobWin** [comment]
I think Swagger2Markup can only process Maps as a body parameter.

**Comment by nikit-cpp** [comment]
@RobWin
What if we assume 'array' type if not specified?
Will it correct?

**Comment by RobWin** [comment]
@nikit-cpp Btw. If you like and use Springfox and Swagger2Markup, we would love to get your GitHub stars :)
